lib/Sema/TypeCheckType.cpp

```diff
@@ -1212,6 +1212,56 @@ static Type diagnoseUnknownType(TypeResolution resolution,
   return ErrorType::get(ctx);
 }
 
+static bool shouldPreferModulesForTopLevelIdentTypeComponent(
```

Comment:
Can this become a method on TypeResolutionOptions, if you added some more state to that class?

Reply:
Sure. Is that better, though?

Reply:
I think its better to encapsulate all the variations on lookup semantics there, yeah.